app/models/vote.rb

```diff
@@ -0,0 +1,49 @@
+class Vote < ActiveRecord::Base
+  class UnknownVotable < Exception; end
+
+  belongs_to :user
+  belongs_to :review
+
+  validates :user, presence: true
+  validates_uniqueness_of :user, scope: [:id, :review_id]
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.